enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,486 @@
+has been deemed sufficient for tracking any SLI impact for `ValidatingAdmissionWebhooks` that we add.
+
+As we are validating at both the webhook and controller (in case webhook is down), we will have separately
+named metrics (i.e. "..._webhook_total" and "..._controller_total") for the same logical items in the above list.
+Given the controller and webhook will be in separate Pods and have separate metric service endpoints registering metrics
+with prometheus, that is the implementation reality.  Any alerts defined can take into account both _webhook and _controller
+metrics of the same logical significance.
+
+Alerts:  using the `rate()` prometheus function, we can compare our new metrics above along with existing metrics from 4.10
+wrt invalid names and failed/successful mounts.  We'll define alerts where, for example, if failures / successes over 1 minute > 0.1, fire.  A pseudo-code
+example: `rate(read_only_failures[1m] / rate(successful_mount[1m]) > 0.1`.  Debating info only to moderate priority for the alert.
+We'll finalize during implementation and update EP accordingly.
+
+Alert 2: if any Shared Resources have an invalid Reference.  Info level priority for user created Shared Resources.  Considering
+moderate to higher priority for pre-populated Shared Resources from other OCP operators.  Or define additional keys to be
+added to the allow list ConfigMap that specifies the priority desired by the component which is creating the pre-populated
+Shared Resource.
+
+#### Impact of existing API Server SLIs
+
+The 'readOnly: true' and 'openshift-' checks involve simple pattern matching on the API objects the webhook(s) are configured
+to validate, so the impacts on API throughput/scalability/availability are thought to be negligible.  In particular, 
+no additional RPCs are envisioned while executing the validation.
+
+The "handling" of 4.10 level Shared Resources created with names starting with "openshift-" will be done in the controller
+and not in the ValidatingAdmissionWebhook.
+
+The controller's examination of the Validity condition in its existing Shared Resource reconciliation loop should 
+have minimal/negligible impact there.  And the controller's removal of data is a behavior whose path length is 
+already established in 4.10, as we remove data if the service account has its 'use' permissions on the Shared Resource
+revoked.
+
+#### Measuring / Verifying impact on existing API Server SLIs
+
+From existing metric discovery, what we will depend on:
+- the EP template example in "Support Procedures" of using `webhook_admission_duration_seconds("<name of our new admission controller>", "validating", "put", "false")`, with >1s latency resulting in the `WebhookAdmissionLatencyHigh` alert firing, appears to be the valid choice addressing our new webhooks' impact on the API Object create/modify flows.  Is that correct?
```

Comment:
This feels correct - if our admission controller has high latency, we fire an alert.